src/community/oseo/oseo-core/src/main/java/org/geoserver/opensearch/eo/store/JDBCOpenSearchAccess.java

```diff
@@ -1135,6 +1135,17 @@ public FeatureStore<FeatureType, Feature> getCollectionSource() throws IOExcepti
         return new JDBCCollectionFeatureStore(this, collectionFeatureType);
     }
 
+    @Override
+    public SimpleFeatureSource getDelegateProductSource() throws IOException {
+        JDBCProductFeatureStore store = new JDBCProductFeatureStore(this, productFeatureType);
+        return store.getDelegateSource();
+    }
+
+    @Override
+    public String mapPropertyNameToSource(String propertyName) {
+        return propertyMapper.getSourceName(propertyName);
+    }
+
```

Comment:
Not sure this is the way to go... it's exposing internals so that the aggregate function can do its job. 
I believe it should work the other way around:
* The function calls the product source, treating it as a regular FeatureSource
* Then it gets a feature collection out of it
* Then it passes a Visitor to the visit method. 
* MappingFeatureCollection handles the visits by unmapping the property name and delegating down

Reply:
fixed